datalake_scripts/datalake.py

```
@@ -0,0 +1,110 @@
+class Datalake:
+
+    authorized_atom_value = [
+        'apk',
+        'as',
+        'cc',
+        'crypto',
+        'cve',
+        'domain',
+        'email',
+        'file',
+        'fqdn',
+        'iban',
+        'ip',
+        'ip_range',
+        'paste',
+        'phone_number',
+        'regkey',
+        'ssl',
+        'url',
+    ]
+
+    def __init__(self, username: str, password: str, env=None):
+        if not env:
+            self.env = 'prod'
+        else:
+            self.env = env
+        self.username = username
+        self.password = password
+        args = ConfigArg(loglevel=logging.WARNING, env=self.env)
+        self.starter = BaseScripts()
+        endpoint_config, _, tokens = self.starter.load_config(
+            args=args, username=self.username, password=self.password)
+        self._threats_api = Threats(
+            endpoint_config, args.env, tokens)
+        self._lookup_threats_api = LookupThreats(
+            endpoint_config, args.env, tokens)
+        self._bulk_lookup_threats_api = BulkLookupThreats(
+            endpoint_config, args.env, tokens)
+        self._post_engine_atom_values_extractor = AtomValuesExtractor(
+            endpoint_config, args.env, tokens)
+        
+    def lookup_threat(self, threat, atom_type=None, hashkey_only=False) -> list:
+        """
+        Use to look up a threat in API.
+
+        :param threat: threat that needs to be looked up.
+        :param atom_type: must be one of the authorized_atom_value
+        """
+        if not atom_type:
+            threats = [threat]
+            atoms_values_extractor_response = self._post_engine_atom_values_extractor.atom_values_extract(
+                threats)
+            if atoms_values_extractor_response['found'] > 0:
+                atom_type = list(
+                    atoms_values_extractor_response['results'].keys())[0]
+            else:
+                logger.warning('your atom could not be typed')
+                exit(1)
```

We can't have `exit` in the library. You'll need to raise an exception (generic or custom).

---

Logs should also only be used to debug, not to communicate errors to users.